File: push_sale_mobile-master/lib/models/pricelist_item.dart
import 'package:push_sale/models/variant_original.dart';

class PriceListItem {
  final int id;
  final int pricelist_id;
  final Variant? variant;
  String sku;
  double price;
  bool updated = false;
  PriceListItem({
    required this.id,
    required this.pricelist_id,
    required this.variant,
    required this.sku,
    required this.price,
  });

  static PriceListItem fromMap(Map<String, dynamic> value) {
    return PriceListItem(
      id: int.tryParse(value["id"]?.toString() ?? "0") ?? 0,
      pricelist_id: int.tryParse(value["pricelist_id"]?.toString() ?? "0") ?? 0,
      variant:
          value["variant"] != null ? Variant.fromMap(value["variant"]) : null,
      sku: value["sku"]?.toString() ?? "",
      price: double.tryParse(value["price"]?.toString() ?? "0") ?? 0,
    );
  }

  static List<PriceListItem> fromListMap(List<dynamic> value) {
    List<PriceListItem> list = [];
    for (var item in value) {
      PriceListItem price = PriceListItem.fromMap(item);
      list.add(price);
    }
    return list;
  }

  Map<String, dynamic> toMap() {
    return {
      "id": id,
      "sku": sku,
      // "variant_id": variant.id,
      "price": price,
      // "pricelist_id": pricelist_id,
    };
  }
}

File: push_sale-master/app/Models/PriceListItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Laravel\Passport\HasApiTokens;

class PriceListItem extends Model
{
    use HasApiTokens, HasFactory;
    protected $table = "pricelist_item";

    protected $fillable = [
        "id",
        "sku",
        "price",
        "pricelist_id",
        "variant_id",
    ];

    protected $hidden = [/*"pricelist_id",  */ "variant_id", "created_at", "updated_at", "deleted_at"];

    public function scopeForDistributor($query, $distributorId)
    {
        return $query->whereHas("pricelist", function ($q) use ($distributorId) {
            $q->where("distributor_id", $distributorId);
        });
    }

    static public function adjuster($list, $distributorId = null){
        $allowedIds = null;
        if ($distributorId) {
            $ids = array_values(array_unique(array_filter(array_map(function ($item) {
                return isset($item["id"]) ? $item["id"] : null;
            }, $list))));

            $allowedIds = array_flip(
                PriceListItem::forDistributor($distributorId)
                    ->whereIn("id", $ids)
                    ->pluck("id")
                    ->all()
            );
        }

        foreach($list as $item){
            if ($allowedIds !== null && !isset($allowedIds[$item["id"]])) {
                continue;
            }
            PriceListItem::where("id",$item["id"])->update(["price"=>$item["price"]]);
        }
    }


    public function variant()
    {
        return $this->hasOne(Variant::class, "id", "variant_id");
    }

    public function pricelist()
    {
        return $this->belongsTo(PriceList::class, "pricelist_id");
    }
}
